src/rooms/room_details/v1.rs

```diff
@@ -0,0 +1,122 @@
+//! [GET /_synapse/admin/v1/rooms/:room_id](https://github.com/element-hq/synapse/blob/master/docs/admin_api/rooms.md#room-details-api)
+use ruma::{
+    api::{metadata, request, response, Metadata},
+    events::room::{guest_access::GuestAccess, history_visibility::HistoryVisibility},
+    room::RoomType,
+    space::SpaceRoomJoinRule,
+    OwnedMxcUri, OwnedRoomAliasId, OwnedRoomId, OwnedUserId, UInt,
+};
+
+const METADATA: Metadata = metadata! {
+    method: GET,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/rooms/:room_id",
+    }
+};
+
+#[request]
+pub struct Request {
+    /// Alias or ID of the room to join.
+    #[ruma_api(path)]
+    pub room_id: OwnedRoomId,
+}
+
+#[response]
+pub struct Response {
+    /// Room ID
+    pub room_id: OwnedRoomId,
+
+    /// Room name
+    pub name: Option<String>,
+
+    /// Room topic
+    pub topic: Option<String>,
+
+    /// Room avatar
+    pub avatar: Option<OwnedMxcUri>,
+
+    /// Room alias ID
+    pub canonical_alias: Option<OwnedRoomAliasId>,
+
+    /// Amount of joined members.
+    pub joined_members: UInt,
+
+    /// Amount of local members.
+    pub joined_local_members: UInt,
+
+    /// A mount of local devices.
```

Comment:
Now there is an extra space.

```suggestion
    /// Amount of local devices.
```